Added patient management documentation (#11)
icure/medtech-docs: sdks/how-to/how-to-manage-patients.md

			"use": "official"
		},
		{
			"firstNames": [
				"Professor"
			],
			"prefix": [],
			"suffix": [],
			"use": "nickname"
		}
	],
	"languages": [],
	"addresses": [],
	"mergedIds": {},
	"active": true,
	"deactivationReason": "none",
	"partnerships": [],
	"patientHealthCareParties": [],
	"patientProfessions": [],
	"parameters": {},
	"properties": {},
	"rev": "2-9da777d089dc3a159d76ab29fff2acd2",
	"created": 1664530613307,
	"modified": 1664530613678,
	"author": "b36fa6cb-d7a8-40f0-bcf6-af6ce0decb78",
	"responsible": "ab623d88-baed-40b9-91b7-ab26e9a08db5",
	"firstName": "Hubert",
	"lastName": "Farnsworth",
	"gender": "male",
	"birthSex": "male",
	"personalStatus": "unknown",
	"dateOfBirth": 28410409,
	"profession": "CEO/Owner of Planet Express, Lecturer at Mars University",
	"note": "Good news everyone!",
	"nationality": "American",
	"systemMetaData": {
		"hcPartyKeys": {},
		"privateKeyShamirPartitions": {},
		"secretForeignKeys": [],
		"cryptedForeignKeys": {},
		"delegations": {
			"ab623d88-baed-40b9-91b7-ab26e9a08db5": {}
		},
		"encryptionKeys": {
			"ab623d88-baed-40b9-91b7-ab26e9a08db5": {}
		},
		"aesExchangeKeys": {},
		"transferKeys": {}
	}
}
```
</details>

> ## How to get a patient&#8239;?

Reviewer: ```suggestion
## How to load a patient&#8239;?
```

> To get a patient, we can use the `getPatient` method on the `PatientApi` object. This method takes one parameter: the patient id.

Reviewer: ```suggestion
To load a patient, we can use the `getPatient` method on the `PatientApi` object. This method takes one parameter: the patient id.
```

	"dateOfBirth": 28410409,
	"profession": "CEO/Owner of Planet Express, Lecturer at Mars University",
	"note": "Good news everyone!",
	"nationality": "American",
	"systemMetaData": {
		"hcPartyKeys": {},
		"privateKeyShamirPartitions": {},
		"secretForeignKeys": [],
		"cryptedForeignKeys": {},
		"delegations": {
			"ab623d88-baed-40b9-91b7-ab26e9a08db5": {}
		},
		"encryptionKeys": {
			"ab623d88-baed-40b9-91b7-ab26e9a08db5": {}
		},
		"aesExchangeKeys": {},
		"transferKeys": {}
	}
}
```
</details>

## How to delete a patient&#8239;?

To delete a patient, we can use the `deletePatient` method on the `PatientApi` object. This method takes one parameter: the patient id.

<!-- file://code-samples/how-to/patients/index.mts snippet:delete a patient-->
```typescript
const deletedPatient = await api.patientApi.deletePatient(patient.id!)
```

<details>
    <summary>Output</summary>

```json
// TODO Add output
```
</details>

## How to filter patients&#8239;?

To filter patients, we can use the `filterPatients` method on the `PatientApi` object. This method takes one parameter: the filter.

<!-- file://code-samples/how-to/patients/index.mts snippet:get a list of patient-->
```typescript

> const filter = await new PatientFilter().forDataOwner(loggedUser.healthcarePartyId!).byGenderEducationProfession('male').dateOfBirthBetween(28000101, 29000101).build()

Reviewer: Isn't this line a bit long ?

const patients = await api.patientApi.filterPatients(filter)
```

<details>
    <summary>Output</summary>

```json
```
</details>

### Filter builder

To create a filter, we can use the [`PatientFilter`](/sdks/references/classes/PatientFilter#methods-1) builder methods. This builder allows us to create complex filter object.

In the example above, we created the filter this way:

<!-- file://code-samples/how-to/patients/index.mts snippet:filter builder-->
```typescript
new PatientFilter().forDataOwner(loggedUser.healthcarePartyId!).byGenderEducationProfession('male').dateOfBirthBetween(28000101, 29000101).build()
```

The resulting filter object will create a filter that allows us to get all `Patient` that satisfy all the following requirements:

> - The `Patient` is owned by the logged user healthcare party

Reviewer: ```suggestion
- The `Patient` is accessible by the currently logged healthcare professional
```
